### DISCUSSION TITLE
On adding support to a new plugin (eduplay)

### DISCUSSION THREAD

**Discussion by cgomesu** [streamlink/streamlink#3890] [Q&A] [answered]
I have recently learned about a new streaming platform ([eduplay](https://eduplay.rnp.br/portal/home)) associated with the Brazilian Ministry of Science, Technology, and Innovation ([MCTI](https://www.gov.br/mcti/)) that provides free content related to education and research in Brazil. I took at look at their platform and did not find any restrictions that would prohibit the use of `streamlink` to access their content and at first glance, it seems pretty straightforward to parse the stream data from their servers.  Here's an example from one of their live streams:

![eduplay-stream-data](https://user-images.githubusercontent.com/50534116/127189466-3ab0512d-3cf1-4125-a043-ed6cd3b02a2b.png)

I've just read the [CONTRIBUTING.md](https://github.com/streamlink/streamlink/blob/d240704d1237fb5878960480c3f8951e0c5023b9/CONTRIBUTING.md) guide and have a few questions related to adding new plugins:

1. Would you rather [receive a proper request to support eduplay](https://github.com/streamlink/streamlink/blob/d240704d1237fb5878960480c3f8951e0c5023b9/CONTRIBUTING.md#plugin-requests) or should I go ahead and [put together a PR that add support to it via a new `streamlink` plugin](https://github.com/streamlink/streamlink/blob/d240704d1237fb5878960480c3f8951e0c5023b9/CONTRIBUTING.md#pull-requests)?
2. In case of submitting a PR, is there a particular development workflow that I should be aware of that was not documented in the [CONTRIBUTING.md](https://github.com/streamlink/streamlink/blob/d240704d1237fb5878960480c3f8951e0c5023b9/CONTRIBUTING.md) guide?  I'm talking about style guides you adhere (PEP) and writing/editing tests for the plugin.  Regarding the latter, I noticed the ones in the `tests/` dir and they look pretty straightforward (they just test whether the URL is correct).
3. Are all PRs submitted directly to the master branch?

**Comment by bastimeyer** [comment, marked as answer]
1. I haven't looked at the site yet, but you could open a plugin request first and wait for the responses first before starting to write and submit a PR which could potentially be closed without being merged, so your time won't be wasted in this case.
2. The code needs to pass the flake8 linting rules and when adding a new plugin, tests need to be added which at least match the plugin's URL(s), as well as an entry in the doc's plugin matrix:
   - https://streamlink.github.io/developing.html
   - https://github.com/streamlink/streamlink/blob/d240704d1237fb5878960480c3f8951e0c5023b9/.github/workflows/main.yml
3. Yes, Streamlink does only have one single branch and doesn't use something like Gitflow, etc.

**Reply by cgomesu** [reply]
Thanks for the prompt replay, @bastimeyer . Will follow your recommendation and open a plugin request first then.

**Reply by gravyboat** [reply]
@cgomesu Is this content actually live? I can't check the site right now and don't want you to waste time if it's just going to be dropped due to the plugin guidelines: https://github.com/streamlink/streamlink/blob/master/CONTRIBUTING.md#plugin-requests

**Reply by cgomesu** [reply]
@gravyboat , yes, the platform **contains live streams** as well as vod and audio.  I opened a plugin request at #3891 where you can find all the relevant information about it.
